source/routing/raptor_api.cpp

```diff
@@ -1145,26 +1146,44 @@ pbnavitia::Response make_isochrone(RAPTOR &raptor,
     return pb_creator.get_response();
 }
 
+static void coord_to_string(std::stringstream& ss,
+                            const double& lon,
+                            const double& lat) {
+    ss << std::setprecision(15) << "[" << lon << "," << lat << "]";
+}
+
+static void ring_to_string(std::stringstream& ss,
+                           const std::vector<type::GeographicalCoord>& ring) {
+    auto it = ring.begin();
+    const auto end = ring.end();
+    if (it != end) { coord_to_string(ss, it->lon(), it->lat()); ++it; }
+    for (; it != end; ++it) {
+        ss << ",";
+        coord_to_string(ss, it->lon(), it->lat());
+    }
+}
+
 static void add_graphical_isochrone(const type::MultiPolygon& shape, PbCreator& pb_creator) {
-    auto pb_isochrone = pb_creator.add_graphical_isochrones();
-    auto pb_polys = pb_isochrone->mutable_geojson();
-    for (const auto& polygon: shape) {
-        auto p = pb_polys->add_polygons();
-        auto lo = p->mutable_outer();
-        for (const auto& coord: polygon.outer()) {
-            auto c = lo->add_coordinates();
-            c->set_lon(coord.lon());
-            c->set_lat(coord.lat());
+    std::stringstream geojson;
```

Comment:
just to remember : you plan on adding the remark of @TeXitoi if I understood well